mmv1/products/cloudrunv2/api.yaml

```
@@ -0,0 +1,715 @@
+                              Port number to access on the container. Must be in the range 1 to 65535. If not specified, defaults to 8080. 
+          - !ruby/object:Api::Type::Array
+            name: "volumes"
+            description: |-
+              A list of Volumes to make available to containers.
+            item_type: !ruby/object:Api::Type::NestedObject
+              properties:
+                - !ruby/object:Api::Type::String
+                  name: "name"
+                  required: true
+                  description: |-
+                    Volume's name.
+                - !ruby/object:Api::Type::NestedObject
+                  name: "secret"
+                  description: |-
+                    Secret represents a secret that should populate this volume. More info: https://kubernetes.io/docs/concepts/storage/volumes#secret
+                  # exactly_one_of:
+                  #   - template.0.volumes.0.secret
+                  #   - template.0.volumes.0.cloudSqlInstance
+                  properties:
+                    - !ruby/object:Api::Type::String
+                      name: "secret"
+                      required: true
+                      description: |-
+                        The name of the secret in Cloud Secret Manager. Format: {secret} if the secret is in the same project. projects/{project}/secrets/{secret} if the secret is in a different project.
+                    - !ruby/object:Api::Type::Integer
+                      name: "defaultMode"
+                      description: |-
+                        Integer representation of mode bits to use on created files by default. Must be a value between 0000 and 0777 (octal), defaulting to 0444. Directories within the path are not affected by this setting.
+                    - !ruby/object:Api::Type::Array
+                      name: "items"
+                      description: |-
+                        If unspecified, the volume will expose a file whose name is the secret, relative to VolumeMount.mount_path. If specified, the key will be used as the version to fetch from Cloud Secret Manager and the path will be the name of the file exposed in the volume. When items are defined, they must specify a path and a version.
+                      item_type: !ruby/object:Api::Type::NestedObject
+                        properties:
+                          - !ruby/object:Api::Type::String
+                            name: "path"
+                            required: true
+                            description: |-
+                              The relative path of the secret in the container.
+                          - !ruby/object:Api::Type::String
+                            name: "version"
+                            description: |-
+                              The Cloud Secret Manager secret version. Can be 'latest' for the latest value or an integer for a specific version
+                          - !ruby/object:Api::Type::Integer
```

I think that HCL may not allow octal integers - possibly we need to represent this with a String?

```suggestion
                          - !ruby/object:Api::Type::String
```

---

The API returns decimal integer for this field. I'm inclined to keep this field as integer (this is also how this field is implemented in v1) and add an example of this use case ([config](https://github.com/hashicorp/terraform-provider-google/blob/main/google/resource_cloud_run_service_generated_test.go#L432)). Or we can also implement this as string and add a DSF.

---

If integer works then that's fine with me. 👍